pkg/render/policyrecommendation.go

```diff
@@ -0,0 +1,329 @@
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      PolicyRecommendationName,
+			Namespace: PolicyRecommendationNamespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: ptr.Int32ToPtr(1),
+			Template: *pr.deploymentPodTemplate(),
+		},
+	}
+}
+
+func (pr *policyRecommendationComponent) deploymentPodTemplate() *corev1.PodTemplateSpec {
+	var ps []corev1.LocalObjectReference
+	for _, x := range pr.cfg.PullSecrets {
+		ps = append(ps, corev1.LocalObjectReference{Name: x.Name})
+	}
+
+	volumes := []corev1.Volume{
+		pr.cfg.TrustedBundle.Volume(),
+		pr.cfg.PolicyRecommendationServerCertSecret.Volume(),
+	}
+
+	container := relasticsearch.ContainerDecorateIndexCreator(
+		relasticsearch.ContainerDecorate(pr.controllerContainer(), pr.cfg.ESClusterConfig.ClusterName(),
+			ElasticsearchPolicyRecommendationUserSecret, pr.cfg.ClusterDomain, rmeta.OSTypeLinux),
+		pr.cfg.ESClusterConfig.Replicas(), pr.cfg.ESClusterConfig.Shards())
+
+	return relasticsearch.DecorateAnnotations(&corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        PolicyRecommendationName,
+			Namespace:   PolicyRecommendationNamespace,
+			Annotations: pr.policyRecommendationAnnotations(),
+		},
+		Spec: corev1.PodSpec{
+			Tolerations:        pr.cfg.Installation.ControlPlaneTolerations,
+			NodeSelector:       pr.cfg.Installation.ControlPlaneNodeSelector,
+			ServiceAccountName: PolicyRecommendationName,
+			ImagePullSecrets:   ps,
+			Containers: []corev1.Container{
+				container,
+			},
+			Volumes: volumes,
+		},
+	}, pr.cfg.ESClusterConfig, pr.cfg.ESSecrets).(*corev1.PodTemplateSpec)
+}
+
+func (pr *policyRecommendationComponent) policyRecommendationAnnotations() map[string]string {
+	annotations := pr.cfg.TrustedBundle.HashAnnotations()
+	if pr.cfg.PolicyRecommendationServerCertSecret != nil {
+		annotations[pr.cfg.PolicyRecommendationServerCertSecret.HashAnnotationKey()] = pr.cfg.PolicyRecommendationServerCertSecret.HashAnnotationValue()
+	}
+	return annotations
+}
+
+func (pr *policyRecommendationComponent) serviceAccount() client.Object {
+	return &corev1.ServiceAccount{
+		TypeMeta:   metav1.TypeMeta{Kind: "ServiceAccount", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: PolicyRecommendationServiceAccountName, Namespace: PolicyRecommendationNamespace},
+	}
+}
+
+// allowTigeraPolicyForPolicyRecommendation defines an allow-tigera policy for policy recommendation.
+func allowTigeraPolicyForPolicyRecommendation(cfg *PolicyRecommendationConfiguration) *v3.NetworkPolicy {
+	egressRules := []v3.Rule{
+		{
+			Action:      v3.Allow,
+			Protocol:    &networkpolicy.TCPProtocol,
+			Destination: networkpolicy.KubeAPIServerServiceSelectorEntityRule,
+		},
+		{
+			Action:      v3.Allow,
+			Protocol:    &networkpolicy.TCPProtocol,
+			Destination: ManagerEntityRule,
+		},
+		{
+			Action:      v3.Allow,
+			Protocol:    &networkpolicy.TCPProtocol,
+			Destination: networkpolicy.ESGatewayEntityRule,
+		},
+		{
```

Comment:
I don't think the controller acts as a server that receives requests, so this does not seem necessary to allow.

Reply:
If I remove this rule we get:

```
2023-03-22 19:15:27.199 [WARNING][1] elastic.go 166: Elastic connect failed, retrying attempts=9 error=Head "https://tigera-secure-es-gateway-http.tigera-elasticsearch.svc:9200": dial tcp 10.108.227.184:9200: i/o timeout
2023-03-22 19:15:27.700 [INFO][1] elastic.go 161: Connecting to Elastic
2023-03-22 19:15:33.704 [WARNING][1] elastic.go 166: Elastic connect failed, retrying attempts=8 error=Head "https://tigera-secure-es-gateway-http.tigera-elasticsearch.svc:9200": context deadline exceeded
```

while attempting to connect to elastic